openrec/implicit_model_trainer.py

```diff
@@ -150,24 +155,23 @@ def _evaluate_partial(self, eval_dataset):
             metric_results[evaluator.name] = []
 
         to_be_saved = dict()
+        to_be_saved["num_negatives"] = self._num_negatives
         to_be_saved["users"] = list()
+        to_be_saved["user_items"] = dict()
+        to_be_saved["results"] = dict()
         for user in tqdm(eval_dataset.get_unique_user_list()):
             to_be_saved["users"].append(int(user))
             items = self._sampled_negatives[user] + eval_dataset.get_interactions_by_user_gb_item(user)
+            to_be_saved["user_items"][int(user)] = items
             scores = self._score_partial_items(user, items)
             result = self._eval_manager.partial_eval(pos_scores=scores[self._num_negatives:], neg_scores=scores[:self._num_negatives])
+            to_be_saved["results"][int(user)] = scores
             for key in result:
                 metric_results[key].append(result[key])
 
         if self._eval_save_prefix:
-            for evaluator in self._eval_manager.evaluators:
-                to_be_saved[evaluator.name] = list(metric_results[evaluator.name])
-                if len(to_be_saved[evaluator.name]) > 0 and type(to_be_saved[evaluator.name][0]) != np.float64:
-                    to_be_saved[evaluator.name] = [list(a) for a in to_be_saved[evaluator.name]]
-                else:
-                    to_be_saved[evaluator.name] = [float(a) for a in to_be_saved[evaluator.name]]
-            tmpf = open(self._eval_save_prefix + "_evaluate_partial.json", 'w')
-            tmpf.write(json.dumps(to_be_saved))
+            tmpf = open(self._eval_save_prefix + "_evaluate_partial.pickle", 'wb')
```

Comment:
It's a better practice to use:

`with open(self._eval_save_prefix + "_evaluate_partial.pickle", 'wb') as tmpf:`
`        pickle.dump(to_be_save, tmpf)`

Reply:
An indentation should be added to the second line...